src/neo/Oracle/Protocols/Https/OracleHttpsProtocol.cs

```diff
@@ -0,0 +1,178 @@
+        /// <returns>Oracle result</returns>
+        public OracleResponse Process(OracleHttpsRequest request)
+        {
+            LoadConfig();
+
+            if (!AllowPrivateHost && IsPrivateHost(Dns.GetHostEntry(request.URL.Host)))
+            {
+                // Don't allow private host in order to prevent SSRF
+
+                return OracleResponse.CreateError(request.Hash, OracleResultError.PolicyError);
+            }
+
+            Task<HttpResponseMessage> result;
+            using var handler = new HttpClientHandler();
+            // TODO: Accept all certificates
+            handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+            using var client = new HttpClient(handler);
+
+            switch (request.Method)
+            {
+                case HttpMethod.GET:
+                    {
+                        result = client.GetAsync(request.URL);
+                        break;
+                    }
+                //case HttpMethod.POST:
+                //    {
+                //        result = client.PostAsync(httpRequest.URL, new ByteArrayContent(httpRequest.Body));
+                //        break;
+                //    }
+                //case HttpMethod.PUT:
+                //    {
+                //        result = client.PutAsync(httpRequest.URL, new ByteArrayContent(httpRequest.Body));
+                //        break;
+                //    }
+                //case HttpMethod.DELETE:
+                //    {
+                //        result = client.DeleteAsync(httpRequest.URL);
+                //        break;
+                //    }
+                default:
+                    {
+                        return OracleResponse.CreateError(request.Hash, OracleResultError.PolicyError);
+                    }
+            }
+
+            try
+            {
+                if (!result.Wait(TimeOut))
+                {
+                    return OracleResponse.CreateError(request.Hash, OracleResultError.Timeout);
+                }
+            }
+            catch
+            {
+                return OracleResponse.CreateError(request.Hash, OracleResultError.ServerError);
+            }
+
+            if (result.Result.IsSuccessStatusCode)
+            {
+                var ret = result.Result.Content.ReadAsStringAsync();
+
+                if (!ret.Wait(TimeOut))
+                {
+                    return OracleResponse.CreateError(request.Hash, OracleResultError.Timeout);
+                }
+
+                if (!ret.IsFaulted)
+                {
+                    if (!OracleService.FilterResponse(ret.Result, request.Filter, out string filteredStr))
+                    {
+                        return OracleResponse.CreateError(request.Hash, OracleResultError.FilterError);
+                    }
+
+                    return OracleResponse.CreateResult(request.Hash, filteredStr);
+                }
```

Comment:
If possible, could we optimize here, as there are too many if branchs to find the normal statement. It's just my personal habit of putting all the error handling together and putting the normal processing at the end. ☺️